pandas/tests/indexes/common.py

```diff
@@ -1058,3 +1059,20 @@ def test_putmask_with_wrong_mask(self):
 
         with pytest.raises(ValueError):
             index.putmask('foo', 1)
+
+    def test_astype_category(self):
```

Comment:
can you pass a ``CategoricalDtype`` here as well (with and w/o ordered) and make this parameterized